src/webapp/components/data-submission/UploadRis.tsx

```diff
@@ -1,100 +1,111 @@
+    const risFileUploadRef = useRef<DropzoneRef>(null);
+
+    const openFileUploadDialog = useCallback(async () => {
+        risFileUploadRef.current?.openDialog();
+    }, [risFileUploadRef]);
 
     useEffect(() => {
-        if (fileList?.length) {
+        if (risFile) {
             validate(true);
         } else {
             validate(false);
         }
-    }, [fileList, validate]);
-
-    const inputRef = useRef<HTMLInputElement>(null);
-
-    const files = fileList ? [...fileList] : [];
-
-    const updateFiles = (e: ChangeEvent<HTMLInputElement>) => {
-        setFileList(e.target.files);
-    };
-
-    const selectFile = () => {
-        if (inputRef.current != null) {
-            inputRef.current.click();
-        }
-    };
+    }, [risFile, validate]);
 
     const removeFiles = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
         event.preventDefault();
-        setFileList(null);
+        setRisFile(null);
     };
 
-    const uploadFiles = () => {
-        if (!fileList) {
-            return;
-        }
-        // Create new FormData object and append files
-        const data = new FormData();
-        files.forEach((file, i) => {
-            data.append(`file-${i}`, file, file.name);
-        });
-
-        // TEST: Uploading the files using the fetch API to mock bin server
-        fetch("https://httpbin.org/post", {
-            method: "POST",
-            body: data,
-        })
-            .then(res => res.json())
-            .then(data => console.debug(data))
-            .catch(err => console.error(err));
-    };
+    const risFileUpload = useCallback(
+        async (files: File[], rejections: FileRejection[]) => {
+            if (rejections.length > 0) {
+                snackbar.error(i18n.t("Multiple uploads not allowed, please select one file"));
+            } else {
+                const uploadedRisFile = files[0];
+                if (uploadedRisFile) {
+                    setIsLoading(true);
+                    setRisFile(uploadedRisFile);
+                    try {
+                        const fileId = await uploadFile(api, uploadedRisFile);
+                        const submission = {
+                            id: generateUid(),
+                            batchId: "Dataset 1",
+                            call: "",
+                            countryCode: "",
+                            fileId,
+                            fileName: uploadedRisFile.name,
+                            fileType: "RIS",
+                            inputLineNb: "",
+                            outputLineNb: "",
+                            module: "",
+                            period: "",
+                            specimens: [],
+                            status: "uploaded",
+                            submissionDate: new Date().toISOString(),
+                        };
+                        await saveToDataStore({ api, key: "submissions", object: submission });
```

Comment:
A component should invoke to a use case and this to a repository. the component should be agnostic of where the data is saved